src/modules.cpp

```diff
@@ -838,6 +839,25 @@ PVOID WINAPI DetourFindPayloadEx(_In_ REFGUID rguid,
     return NULL;
 }
 
+BOOL WINAPI DetourFreePayload(_In_ PVOID pvData)
+{
+	BOOL fSucceeded = FALSE;
+
+	HMODULE hModule = DetourGetContainingModule(pvData);
+	assert(hModule != NULL);
+	if (hModule != NULL)
+	{
+		fSucceeded = VirtualFree(hModule, 0, MEM_RELEASE);
+		assert(fSucceeded);
+		if (fSucceeded)
+		{
+			hModule = NULL;
```

Comment:
Not sure why this is needed - it's a local variable. I would suggest

```cpp
BOOL WINAPI DetourFreePayload(_In_ PVOID pvData)
{
	BOOL fSucceeded = FALSE;

	HMODULE hModule = DetourGetContainingModule(pvData);
	if (hModule != NULL)
	{
		fSucceeded = VirtualFree(hModule, 0, MEM_RELEASE);
	}

	return fSucceeded;
}

Reply:
Reset pointer variable is a good habit for program, and it is not important. It won't affect the rest of the place.

Reply:
The latest commit has resolved it.